src/context.c

```diff
@@ -140,6 +140,39 @@ static gchar* get_cmdline_bootname(void)
 	return bootname;
 }
 
+static gchar* get_custom_bootname(void)
+{
+	g_autoptr(GSubprocessLauncher) launcher = NULL;
+	g_autoptr(GSubprocess) handle = NULL;
+	g_autoptr(GDataInputStream) datainstream = NULL;
+	g_autoptr(GPtrArray) args_array = NULL;
+	g_autofree gchar *outline = NULL;
+	GInputStream *instream;
+
+	args_array = g_ptr_array_new();
+	g_ptr_array_add(args_array, context->config->custom_bootloader_backend);
+	g_ptr_array_add(args_array, "get-primary");
```

Comment:
'primary' in our API is the slot (bootname) to be booted next.

Once you have installed an update or manually marked a slot active, this information will change and subsequent calls of this function without a reboot will give incorrect results.

If you have an explicit method that provides the information about the currently booted slot, it would be better to add an explicit API call in RAUC's bootchooser implementation (like `get-current`). For example, UEFI should already provide the current booted entry in the `BootCurrent` variable.
Maybe using this is an option, too?.

Reply:
> If you have an explicit method that provides the information about the currently booted slot, it would be better to add an explicit API call in RAUC's bootchooser implementation (like get-current).

I like this idea, but the `custom_backend_get` uses `r_context` internally, which is not available in context.c. Should I add a minimal method like the one I proposed?

> For example, UEFI should already provide the current booted entry in the BootCurrent variable.

Not this UEFI, this provides `781e084c-a330-417c-b678-38e696380cb9-BootChainOsCurrent`

Reply:
> I like this idea, but the custom_backend_get uses r_context internally, which is not available in context.c. 

I see.

> Should I add a minimal method like the one I proposed?

Having it in bootchooser would be even better, but I guess we could also accept it in context only first of all.

However, the commit should then also touch the [Custom bootloader backend interface](https://rauc.readthedocs.io/en/latest/integration.html#custom-bootloader-backend-interface) documentation and mentions the purpose of `get-current`.

> Not this UEFI, this provides 781e084c-a330-417c-b678-38e696380cb9-BootChainOsCurrent

Oh, interesting. I've assumed this would be a standard var so far.